specification/security/resource-manager/Microsoft.Security/preview/2020-08-06-preview/insights.json

```diff
@@ -0,0 +1,187 @@
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{resourceId}/providers/Microsoft.Security/insights/{insightName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get classification insights of an Azure storage resource": {
+            "$ref": "./examples/Insights/GetStorageClassificationInsights_example.json"
+          }
+        },
+        "tags": [
+          "Insights"
+        ],
+        "description": "Insights of a specific resource",
+        "operationId": "Insights_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/InsightName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/InsightsModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InsightsList": {
+      "type": "object",
+      "readOnly": true,
+      "description": "Insights list response.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of insights.",
+          "items": {
+            "$ref": "#/definitions/InsightsModel"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "InsightsModel": {
+      "type": "object",
+      "description": "Insights of a specific resource.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Insights details data",
+          "$ref": "#/definitions/InsightsProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "InsightsProperties": {
+      "type": "object",
+      "description": "Describes properties of resource insights.",
+      "properties": {
+        "associatedResource": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Azure resource ID of the associated resource."
+        },
+        "insightDescription": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Describes the insight."
+        },
+        "insightProperties": {
+          "readOnly": true,
+          "type": "object",
```

Comment:
This modeling makes it impossible for many clients to deserialize and work with this data.  Is this a polymorphic type with different possible formats?  Is there a set of known properties and then an additional property bag which is designed to be unstructured?  From the look of the examples, these appear to be fairly structured.

Reply:
Hi Mark, this is a polymorphic type - though we currently support only one type of insight (as indicated in the example), we plan to follow this up with other types in the future, so this is by design.